### PR TITLE
fix: not supported Powershell and Python versions on Automation Runbook type

### LINKED ISSUE #1445: [AVM Module Issue]: Runbook module's PowerShell version
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/automation/automation-account

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

The runbook module, when configured for PowerShell, defaults to 5.1 without an option to specify. This is not changeable within the portal, making this type of AVM deployment limited in usefulness.

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #1743: [AVM Module Issue]: Add support for PowerShell Core 'PowerShell72' as a Automation Runbook type
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/automation/automation-account

### (Optional) Module Version

0.4.1

### Description

Creation of 'PowerShell core' runbook is now supported through Azure Bicep, refer [runbookType](https://learn.microsoft.com/en-us/azure/templates/microsoft.automation/automationaccounts/runbooks?pivots=deployment-language-bicep#runbookcreateorupdatepropertiesorrunbookproperties)

The AVM module for Automation Account, does not support **"PowerShell72"** as a [runbook type](https://github.com/Azure/bicep-registry-modules/tree/main/avm/res/automation/automation-account/runbook#parameter-type). The allowed values at the moment are as below and the module fails when 'PowerShell72' is supplied as a type

**_'The provided value for the template parameter 'type' is not valid. The value 'PowerShell72' is not part of the allowed value(s):_**

[
  'Graph'
  'GraphPowerShell'
  'GraphPowerShellWorkflow'
  'PowerShell'
  'PowerShellWorkflow'
]

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #1757: [AVM Module Issue]: automation-account/runbook allow Python3 in type parameter
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/automation/automation-account

### (Optional) Module Version

_No response_

### Description

Within the module for automation-account/runbook, the type parameters are limited to - 
```json
[
  'Graph'
  'GraphPowerShell'
  'GraphPowerShellWorkflow'
  'PowerShell'
  'PowerShellWorkflow'
]
```
... however, the allowed values [here on microsoft.com](https://learn.microsoft.com/en-us/azure/templates/microsoft.automation/automationaccounts/runbooks?pivots=deployment-language-bicep) are - 
```json
[
  'Graph'
  'GraphPowerShell'
  'GraphPowerShellWorkflow'
  'PowerShell'
  'PowerShell72'
  'PowerShellWorkflow'
  'Python2'
  'Python3'
  'Script' 
]
```
Would it be possible to add the rest in? Or at least the Python ones? Apologies if there is more under the hood that is preventing this from being added that I haven't considered.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by elanzel**
## Description
- updated runbook API version t 2023-11-01 to support all the following runbook types:
'Graph'
'GraphPowerShell'
'GraphPowerShellWorkflow'
'PowerShell'
'PowerShell72'
'PowerShellWorkflow'
'Python2'
'Python3'
'Script' (required)
- updated useridentity and keyservices to be compliant with new bicep requirements
- updated PSRule excemption file for default checks

Closes #1757 
Closes #1743 
Closes #1445 

## Pipeline Reference

| Pipeline |
| -------- |
[avm.res.automation.automation-account](https://github.com/elanzel/bicep-registry-modules/actions/workflows/avm.res.automation.automation-account.yml)
|          |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [X] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/automation/automation-account/main.bicep [hunk 1]
```diff
@@ -172,15 +172,15 @@ resource automationAccount 'Microsoft.Automation/automationAccounts@2022-08-08'
     }
     encryption: !empty(customerManagedKey)
       ? {
-          keySource: 'Microsoft.KeyVault'
+          keySource: !empty(customerManagedKey) ? 'Microsoft.Keyvault' : 'Microsoft.Automation'
```
Comment (AlexanderSehr) [inline avm/res/automation/automation-account/main.bicep:175]: Hey @elanzel, can this case ever happen? 2 lines above you check if the CMK object is not empty - so it should never be possible to actually end up the the Service-Level encryption in this line...

File: avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml [hunk 2]
```diff
@@ -15,6 +15,8 @@ spec:
     - Azure.Cognitive.ManagedIdentity
       # Automation specific
     - Azure.Automation.ManagedIdentity
+    - Azure.Automation.AuditLogs
+    - Azure.Automation.PlatformLogs
```
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml:19]:
```suggestion
    - Azure.Automation.AuditLogs # Diagnostic Settings cannot be set by default, but require user input
    - Azure.Automation.PlatformLogs # Diagnostic Settings cannot be set by default, but require user input
```

Comment (AlexanderSehr) [reply avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml:19]: We should probably start doing this 😄

File: avm/res/automation/automation-account/main.bicep (hunk 1 above)
Comment (elanzel) [reply avm/res/automation/automation-account/main.bicep:175]: i needed to do it to solve this
 Warning BCP036: The property "keySource" expected a value of type "'Microsoft.Automation' | 'Microsoft.Keyvault' | null" but the provided value is of type "'Microsoft.KeyVault'". If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]

Comment (AlexanderSehr) [reply avm/res/automation/automation-account/main.bicep:175]: In that case, I have good news ;) 
```suggestion
          keySource: 'Microsoft.Keyvault'
```

The RP just messed up when creating the property, not applying the correct casing.

Comment (AlexanderSehr) [reply avm/res/automation/automation-account/main.bicep:175]: And to be super explicit it's this:
- `Microsoft.KeyVault` vs
- `Microsoft.Keyvault`

Comment (AlexanderSehr) [reply avm/res/automation/automation-account/main.bicep:175]: Please test and let me know :)

Comment (elanzel) [reply avm/res/automation/automation-account/main.bicep]: Ohh i didn't noticed it! It works 👍

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**